phat/contracts/action_offchain_rollup/lib.rs

```diff
@@ -234,10 +241,11 @@ mod action_offchain_rollup {
         #[ink(message)]
         pub fn config_core_script(&mut self, code_hash: Hash) -> Result<()> {
             self.ensure_owner()?;
-            let Some(core) = self.core.get() else {
+            let Some(mut core) = self.core.get() else {
                 return Err(Error::CoreNotConfigured);
             };
-            self.config_core_inner(code_hash, core.settings)
+            core.code_hash = code_hash;
+            self.config_core_inner(core)
         }
```

Comment:
Since we have introduced a new struct called `Core`, can we remove this function or rename to `config_core_hash`?

Reply:
The complete name could be `config_core_script_with_code_hash`, but it's too lengthy. Since the parameter `code_hash` already signifies the `hash`, we can exclude `_with_code_hash`.
`config_core_hash` would be more weird because there isn't a concept named `core hash`.

As for removing this function, that depends on whether front end tools use it or not. If the front end doesn't use it, I think it's no harm to keep it there as it doesn't require getting the core config before update implementation, might be a bit valuable in tests.

Reply:
The entire workflow needs to be rewritten for the codebase version. The function will be useful in the update scenario where a user uploads a new script and configures it with a new code hash. That's the reason for renaming it to `config_code_hash`.

Reply:
I believe `config_core_script` is more descriptive than either `config_core_hash` or `config_code_hash`, as it best represents its functionality. Consider the `code hash` as simply a pointer which pointing to a piece of code. 

If you still find this confusing, let me add a commit [here](https://github.com/Phala-Network/phat-bricks/pull/36/commits/e14f4644148d9df2c4e3f3efc072907b5616aee9) to explain and de-confuse it.

Reply:
The newer version looks better to me.